.github/workflows/run-tests.yml

```
@@ -0,0 +1,238 @@
+name: Run tests
+
+on: [pull_request]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup PHP with PECL extension
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.1
+          tools: composer
+        env:
+          fail-fast: true
+
+      - name: Software versions
+        id: versions
+        uses: cafuego/command-output@main
+        with:
+          run: |
+            php --version && composer --version
+
+      - name: Checkout Code
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Composer Validate
+        id: validate
+        uses: cafuego/command-output@main
+        with:
+          run: |
+            composer validate
+        env:
+          fail-fast: true
+
+      - name: Code Lint
+        id: lint
+        uses: cafuego/command-output@main
+        with:
+          run: |
+            test ! -d ./html/modules/custom || find -L ./html/modules/custom -iregex '.*\.\(php\|module\|inc\|install\)$' -print0 | xargs -0 -n 1 -P 4 php -l
+            test ! -d ./html/themes/custom || find -L ./html/themes/custom -iregex '.*\.\(php\|theme\)$' -print0 | xargs -0 -n 1 -P 4 php -l
+        env:
+          fail-fast: true
+
+      - name: Configure AWS Credentials
+        id: aws
+        uses: aws-actions/configure-aws-credentials@v1-node16
```

The node version is apparently fixed in v2.
```suggestion
        uses: aws-actions/configure-aws-credentials@v2
```